pkg/tests/ossm/smcp_install_test.go

```diff
@@ -15,219 +15,244 @@
-		if !strings.Contains(msg, "ComponentsReady") {
-			log.Log.Error("SMCP not Ready")
-			t.Error("SMCP not Ready")
-		}
-		util.Shell(`oc get -n %s pods`, meshNamespace)
-	})
-
-	t.Run("smcp_test_upgrade_2.2_to_2.3", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Create SMCP v2.2 in namespace ", meshNamespace)
-		util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
-		util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV22_template, smcp))
-		util.KubeApplyContents(meshNamespace, smmr)
-
-		// patch SMCP identity if it's on a ROSA cluster
-		if env.Getenv("ROSA", "false") == "true" {
-			util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
-		}
-		log.Log.Info("Waiting for mesh installation to complete")
-		util.Shell(`oc wait --for condition=Ready -n %s smcp/%s --timeout 300s`, meshNamespace, smcpName)
-
-		log.Log.Info("Verify SMCP status and pods")
-		util.Shell(`oc get -n %s smcp/%s -o wide`, meshNamespace, smcpName)
-		util.Shell(`oc get -n %s pods`, meshNamespace)
-
-		log.Log.Info("Upgrade SMCP to v2.2")
-		util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV23_template, smcp))
-
-		// patch SMCP identity if it's on a ROSA cluster
-		if env.Getenv("ROSA", "false") == "true" {
-			util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
-		}
-		log.Log.Info("Waiting for mesh installation to complete")
-		time.Sleep(time.Duration(10) * time.Second)
-		util.Shell(`oc wait --for condition=Ready -n %s smcp/%s --timeout 360s`, meshNamespace, smcpName)
-
-		log.Log.Info("Verify SMCP status and pods")
-		msg, _ := util.Shell(`oc get -n %s smcp/%s -o wide`, meshNamespace, smcpName)
-		if !strings.Contains(msg, "ComponentsReady") {
-			log.Log.Error("SMCP not Ready")
-			t.Error("SMCP not Ready")
+	NewTest(t).Id("A1").Groups(Smoke, Full, InterOp, ARM).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		t.Cleanup(func() {
+			installDefaultSMCP(t, meshNamespace)
+		})
+		vars := vars{
+			Name:      smcp.Name,
+			Namespace: smcp.Namespace,
 		}
-		util.Shell(`oc get -n %s pods`, meshNamespace)
+		t.NewSubTest("smcp_test_install_2.4").Run(func(t TestHelper) {
+			t.LogStep("Crete Namespace and Install SMCP v2.4")
+			oc.CreateNamespace(t, meshNamespace)
+			oc.ApplyTemplate(t, meshNamespace, smcpV24_template, vars)
+			if env.IsRosa() {
+				oc.PatchWithMerge(
+					t, meshNamespace,
+					fmt.Sprintf("smcp/%s", smcpName),
+					`{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}`)
+			}
+			oc.WaitSMCPReady(t, meshNamespace, smcpName)
+			oc.ApplyString(t, meshNamespace, smmr)
+			t.LogStep("Check SMCP is Ready")
+			oc.WaitCondition(t, meshNamespace, "smcp", smcpName, "Ready")
+		})
+		t.NewSubTest("smcp_test_uninstall_2.4").Run(func(t TestHelper) {
```

Comment:
Please add empty lines between subtests, so that each part of the test is easily identifiable.

Reply:
Ok